### PR TITLE
Fixed create seeders for Job opportunities in dev environment

### PR DESCRIPTION
**Pull request by Bahugunajii**
#1638 

### Description 
-Created seeders for Job opportunities in a dev environment

### Checklist
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by kuldeep3** [comment]
@Bahugunajii Could you please fix the issue with CI checks that are not passing and then assign it for review?

**Comment by Bahugunajii** [comment]
> @Bahugunajii Could you please fix the issue with CI checks that are not passing and then assign it for review?

Fixed all CI checks!

**Comment by P4NK4J** [comment]
@Bahugunajii The seeders are throwing error while running the command
image

**Review by P4NK4J — CHANGES_REQUESTED** [review]
requesting changes, as seeders are throwing an error while running the module:seed command

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        $faker = Faker::create();
+
+        return [
+            'id' => $index,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:28]: Remove this

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            'opportunity_id' => $faker->randomDigit,
+            'title' => $faker->title,
+            'description' => $faker->text,
+            'type' => $faker->sentence(2),
+            'domain' => $faker->text,
+            'start_date' => $faker->date,
+            'link' => 'https=>//coloredcow.com/career/laravel-developer/',
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:35]: Make it null

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        $faker = Faker::create();
+
+        return [
+            'id' => $index,
+            'opportunity_id' => $faker->randomDigit,
+            'title' => $faker->title,
+            'description' => $faker->text,
+            'type' => $faker->sentence(2),
+            'domain' => $faker->text,
+            'start_date' => $faker->date,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:34]: Use todays date or make it null.

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            'link' => 'https=>//coloredcow.com/career/laravel-developer/',
+            'end_date' => $faker->date,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:36]: Make it null

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            'facebook_post' => $faker->text,
+            'twitter_post' => $faker->text,
+            'linkedin_post' => $faker->text,
+            'instagram_post' => $faker->text,
+            'created_at' => $faker->date,
+            'updated_at' => $faker->date,
+            'posted_by' => $faker->randomDigit,
+            'status' => $faker->text,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:44]: status available - archived, published, draft

But in this case use published

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            'deleted_at' => $faker->date,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:45]: Make it null

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            'start_date' => $faker->date,
+            'link' => 'https=>//coloredcow.com/career/laravel-developer/',
+            'end_date' => $faker->date,
+            'facebook_post' => $faker->text,
+            'twitter_post' => $faker->text,
+            'linkedin_post' => $faker->text,
+            'instagram_post' => $faker->text,
+            'created_at' => $faker->date,
+            'updated_at' => $faker->date,
+            'posted_by' => $faker->randomDigit,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:43]: If it is nullable then make it null, else give the id of any user that exists in the database.

File: Modules/HR/Database/Factories/HrJobFactory.php [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+            'description' => $faker->text,
+            'type' => $faker->sentence(2),
+            'domain' => $faker->text,
+            'start_date' => $faker->date,
+            'link' => 'https=>//coloredcow.com/career/laravel-developer/',
+            'end_date' => $faker->date,
+            'facebook_post' => $faker->text,
+            'twitter_post' => $faker->text,
+            'linkedin_post' => $faker->text,
+            'instagram_post' => $faker->text,
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobFactory.php:40]: Make it null

File: Modules/HR/Database/Seeders/HRDatabaseSeeder.php [hunk 9]
```diff
@@ -24,5 +24,6 @@ public function run()
         $this->call(HrApplicantsTableSeeder::class);
         $this->call(HrApplicationRoundTableSeeder::class);
         $this->call(HrApplicationsTableSeeder::class);
+        $this->call(HRJobSeederTableSeeder::class);
```
Comment (GauravGusain98) [inline Modules/HR/Database/Seeders/HRDatabaseSeeder.php:27]: Move it after `SettingsTableSeeder` seeder line

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
please check the comments

File: Modules/HR/Database/Factories/HrJobFactory.php (hunk 3 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobFactory.php]: Done!

File: Modules/HR/Database/Factories/HrJobFactory.php (hunk 4 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobFactory.php]: Done!

File: Modules/HR/Database/Factories/HrJobFactory.php (hunk 5 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobFactory.php]: Done!

File: Modules/HR/Database/Factories/HrJobFactory.php (hunk 6 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobFactory.php]: done!

File: Modules/HR/Database/Factories/HrJobFactory.php (hunk 7 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobFactory.php]: Done!

File: Modules/HR/Database/Factories/HrJobFactory.php (hunk 8 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobFactory.php]: Done!

File: Modules/HR/Database/Seeders/HRDatabaseSeeder.php (hunk 9 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Seeders/HRDatabaseSeeder.php]: Done!

File: Modules/HR/Database/Factories/HrJobsFactory.php [hunk 10]
```diff
@@ -25,11 +25,23 @@ public function definition()
         $faker = Faker::create();
 
         return [
-            'domain' => array_rand(config('hr.opportunities.domains')),
-            'status' => array_rand(config('hr.opportunities-status')),
-            'type' => config('hr.opportunities.job.type'),
+            'opportunity_id' => 0,
             'title' => $faker->jobTitle,
             'description' => $faker->text(),
+            'type' => config('hr.opportunities.job.type'),
```
Comment (GauravGusain98) [inline Modules/HR/Database/Factories/HrJobsFactory.php:31]: Please use the job type here

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Minor suggestion. Please check

File: Modules/HR/Database/Factories/HrJobsFactory.php [hunk 11]
```diff
@@ -25,11 +25,23 @@ public function definition()
...
+            'link' => null,
+            'end_date' => null,
+            'facebook_post' => null,
+            'twitter_post' => null,
+            'linkedin_post' => null,
+            'instagram_post' => null,
+            'created_at' => null,
+            'updated_at' => null,
+            'posted_by' => null,
+            'status' => array_rand(config('hr.opportunities-status')),
```
[the remark this answers is not in the document]
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobsFactory.php]: > For now make all the job status as `published`

Changes made!

File: Modules/HR/Database/Factories/HrJobsFactory.php (hunk 10 above)
Comment (Bahugunajii) [reply Modules/HR/Database/Factories/HrJobsFactory.php]: Changes made!

**Review by GauravGusain98 — APPROVED** [review]
Looks good

**Review by P4NK4J — APPROVED** [review]
looks good

**Pull request merged**